src/PowerPing/Ping.cs

```diff
@@ -253,33 +248,27 @@ public void Flood(string address)
             // Disable output for faster speeds
             Display.ShowOutput = false;
 
-            // Start flood thread
-            var thread = new Thread(() => {
-                p.Send(attrs);
-            });
-            thread.IsBackground = true;
-            thread.Start();
-            IsRunning = true;
+            var limiter = new DisplayUpdateLimiter(TimeSpan.FromMilliseconds(500));
+            ulong previousPingsSent = 0;
+            void OnResultsUpdate(PingResults r) {
+                // Make sure we're not updating the display too frequently
+                if (!limiter.RequestUpdate()) {
+                    return;
+                }
 
-            // Results loop 
-            while (IsRunning) {
-                // Update results text
-                Display.FloodProgress(p.Results, address);
+                // Calculate pings per second
+                double? pingsPerSecond = (r.Sent - previousPingsSent) / limiter.ElapsedSinceLastUpdate?.TotalSeconds;
```

Comment:
In this context the question mark is called the null-propagation operator, which makes TotalSeconds return a nullable double instead of throwing a NullReferenceException if ElapsedSinceLastUpdate is null. In turn, it also makes the division calculation result in null.

If I weren't using nullable types here, there would need to be logic to avoid divide by zero since ElapsedSinceLastUpdate would be presumably zero on the very first update.